pipeline/pipeline/task.py

```diff
@@ -111,18 +284,105 @@ def handler(match_obj):
         return self
 
     def label(self, label):
+        """
+        Set the task's label.
+
+        Examples
+        --------
+
+        Set the task's label to `qc`:
+
+        >>> t1 = p.new_task()
+        >>> (t1.label('qc')
+        ...    .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        label: :obj:`str`
+
+        Returns
+        -------
+        :class:`.Task`
+            Same task object with label set.
+        """
         self._label = label
         return self
 
     def memory(self, memory):
+        """
+        Set the task's memory requirements.
+
+        Examples
+        --------
+
+        Set the task's memory requirement to 5GB:
+
+        >>> t1 = p.new_task()
+        >>> (t1.memory(5)
+        ...    .command(f'echo "hello"'))
+
+        Parameters
+        ----------
+        memory: :obj:`str` or :obj:`float` or :obj:`int`
```

Comment:
In GB, yes?

Reply:
I presume string is currently just cast to int, but are you thinking of support for strings like "200MB" etc?

Reply:
Right now it's getting passed directly to docker and kubernetes and rendered with f-strings. So it's whatever they support. I can add better checking/functionality in future, but didn't seem important now.

Reply:
Yep, sounds good. But might just want to note that it's in GB here for the time being